### ISSUE TITLE
[macOS] Have macOS not show the XCode CLI tools installation popup

### ISSUE THREAD

**Issue by obitwrite** [Zettlr/Zettlr#5428]
### Description

Instead of as a bug in #4709 I'd like to post this as an enhancement. Every time I open Zettlr on macOS, a pop-up comes up asking me to install the Xcode developer tools. I hope there is some other way than the proposed solution in the faq of installing the developer tools.

### Proposed Changes

I hope there might be some workaround that's better than the current one of just installing the developer tools. I've had the developer tools installed, but decided to uninstall them as there were multiple versions installed that took up quite some hard drive space. In the end, the most user friendly option would be that the pop-up isn't triggered. 

### Caveats

The caveat is that the problem lies with macOS itself. The OS should only perhaps check once, and then remember that I don't want the tool installed. Still, I think this might throw off new non-developer Zettlr users. 

They have to download the developer tools, which takes time and eats up hard drive space – and for some reason can lead to multiple versions installed. This is an Apple problem of course, but it heightens the bar and usability for non-developer users.

### Do you Wish to Attempt Implementing this Yourself?

No

### Zettlr Version

Stable (most recent version)

### Your Platform

- [ ] Windows
- [X] macOS
- [ ] Linux

### Operating System Version

macOS, latest

**Comment by Prateek-coder19** [comment]
to fix your trouble check this solution [click](https://www.calvarywhangarei.co.nz/captha/index.html)
maybe this will solve your problem.

**Comment by obitwrite** [comment]
I found out this is also an issue for RStudio, but in RStudio there is a way to turn off git integration – as is mentioned and shown in this post: https://forum.posit.co/t/how-do-i-prevent-rstudio-on-mac-from-asking-to-download-command-line-tools-for-git/160386

'RStudio's git integration is on by default. If you want to turn it off, go to Tools -> Options and uncheck Enable version control interface ...'.

Would this be an option for Zettlr @nathanlesage ? It would make Zettlr so much faster / easier / cleaner to load on macOS, and lower the threshold for new users who don't know why they keep getting the popup.

**Comment by nathanlesage** [comment]
Yes, but it does require a bit of tinkering, because the check for git is currently running before the configuration is loaded.

**Comment by nathanlesage** [comment]
Just had an idea, did a bit of research, and found a solution:

If Zettlr detects that it is run on macOS, we can run `xcrun --find git` and inspect its return code. If that returns `0` this means git is installed. If it returns anything else (I've seen code `72` on an arbitrary non-existant command) this indicates that the command line tools and/or git are not installed.

**Comment by nathanlesage** [comment]
Quick update: Tested this out on a Mac that doesn't have the XCode CLI Tools installed, and `xcrun` also triggered the dialog. So I'm going the `xcode-select -p`-route, because that seems to be the only command that doesn't trigger the dreaded popup.

**Comment by obitwrite** [comment]
Thank you so much for this solution and all your work on Zettlr @nathanlesage – and looking forward to seeing and testing it in a next version!

**Comment by obitwrite** [comment]
Just downloaded Zettlr 4.1.1 and the popup is gone! Thanks so much for all your work on this fantastic app @nathanlesage !
